merlin/models/tf/models/base.py

```diff
@@ -398,6 +397,19 @@ def evaluate(
 
             x = BatchedDataset(x, batch_size=batch_size, **kwargs)
 
+        # Check if top-k evaluation is set for RetrievalModel,
+        # if not set it to default BruteForce top-k evaluation
+        # using evaluation_candidates as indices.
+        if isinstance(self, RetrievalModel):
+            self.check_for_retrieval_task()
+            if not self.loss_block.pre_eval_topk:
+                if not self.evaluation_candidates:
+                    raise ValueError(
+                        "You need to specify the set of negatives to use for evaluation "
+                        "via `evaluation_candidates` argument"
+                    )
+                self = self._load_topk_evaluation(self.evaluation_candidates, k=self.k)
```

Comment:
We could maybe have no arguments in `_load_topk_evaluation()` as the arguments are already object (self) attributes.